common/src/main/java/earth/terrarium/hermes/api/defaults/ComponentTagElement.java

```diff
@@ -1,112 +1,18 @@
-                    this.color.getValue(),
-                    this.shadowed
-            );
-
-            if (mouseY >= (y + ySurround + contentHeight) || mouseY < (y + ySurround) || !hovered) {
-                continue;
-            }
-            renderComponentHoverEffect(font.getSplitter().componentStyleAtWidth(lines.get(i), Mth.floor(x - mouseX)));
-        }
-    }
-
-    @Override
-    public void setContent(String content) {
-        try {
-            this.text = Component.Serializer.fromJson(content);
-        } catch (Exception e) {
-            this.text = CommonComponents.EMPTY;
-        }
-    }
-
-    @Override
-    public int getHeight(int width) {
-        int lines = Minecraft.getInstance().font.split(this.text, width - (10 + (2 * xSurround))).size();
-        int lineHeight = Minecraft.getInstance().font.lineHeight;
-        return ((lines * lineHeight) + (lines - 2)) + (2 * ySurround); // element height + vertical spacing
-    }
-
-    @Override
-    public boolean mouseClicked(double mouseX, double mouseY, int button, int width) {
-        if (mouseY >= this.getHeight(width) || mouseY < 0) {
-            return false;
-        }
-        Style style = Minecraft.getInstance().font.getSplitter().componentStyleAtWidth(this.text, Mth.floor(mouseX));
-        return Minecraft.getInstance().screen != null && Minecraft.getInstance().screen.handleComponentClicked(style);
-    }
-
-    protected void renderComponentHoverEffect(@Nullable Style style) {
-        Screen screen = Minecraft.getInstance().screen;
-        if (screen == null) return;
-        if (style != null && style.getHoverEvent() != null) {
-            List<FormattedCharSequence> tooltip = new ArrayList<>();
-            HoverEvent event = style.getHoverEvent();
-            var itemInfo = event.getValue(HoverEvent.Action.SHOW_ITEM);
-            if (itemInfo != null) {
-                for (Component component : Screen.getTooltipFromItem(Minecraft.getInstance(), itemInfo.getItemStack())) {
-                    tooltip.add(component.getVisualOrderText());
-                }
-            } else {
-                var entityInfo = event.getValue(HoverEvent.Action.SHOW_ENTITY);
-                if (entityInfo != null) {
-                    if (Minecraft.getInstance().options.advancedItemTooltips) {
-                        entityInfo.getTooltipLines().forEach((component) -> tooltip.add(component.getVisualOrderText()));
-                    }
-                } else {
-                    Component component = event.getValue(HoverEvent.Action.SHOW_TEXT);
-                    if (component != null) {
-                        tooltip.addAll(Minecraft.getInstance().font.split(component, Math.max(screen.width / 2, 200)));
-                    }
-                }
-            }
-
-            if (!tooltip.isEmpty()) {
-                screen.setTooltipForNextRenderPass(tooltip);
-            }
-        }
+    public void addText(String content) {
```

Comment:
override add children to disallow adding children, as in the future I want to deprecate the component tag fully, also it will be weird if people could do `<component>{"text":"text"}<red>test</red>{"text":"more text"}</component>`

Reply:
Good catch; done with 8d618ff.